src/main/java/org/folio/readingroom/repository/ReadingRoomRepository.java

```diff
@@ -1,10 +1,33 @@
 package org.folio.readingroom.repository;
 
+import java.util.List;
 import java.util.UUID;
 import org.folio.readingroom.domain.entity.ReadingRoomEntity;
 import org.folio.spring.cql.JpaCqlRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
 @Repository
 public interface ReadingRoomRepository extends JpaCqlRepository<ReadingRoomEntity, UUID> {
+  @Query(value = """
+      SELECT r
+        FROM ReadingRoomEntity r
+        LEFT JOIN fetch r.patronPermissions p
+        WHERE (p.userId = :userId OR p.userId IS NULL)
```

Comment:
Why the or condition with UserID IS NULL ?

Reply:
@gurleenkaurbp : 
This is because we are sending response with assigned ReadingRooms and NonAssigned both.